File: ios/MyProject/PythonAdd.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(PythonAdd, NSObject)

RCT_EXTERN_METHOD(add:(nonnull NSNumber *)a
                  b:(nonnull NSNumber *)b
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/MyProject/PythonAdd.swift
import Foundation
import React

@objc(PythonAdd)
class PythonAdd: NSObject {

  @objc
  func add(_ a: NSNumber, b: NSNumber, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    // Same logic as add_numbers.py (iOS has no Python runtime)
    let sum = a.doubleValue + b.doubleValue
    resolve(NSNumber(value: sum))
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
